docs/use-cases/NVIDIA-GPU-passthrough-and-Kata.md

```diff
@@ -0,0 +1,298 @@
+# It conflicts with NVIDIA official driver
+CONFIG_DRM_NOUVEAU=n
+```
+
+> **Note**: `CONFIG_DRM_NOUVEAU` is normally disabled by default.
+It is worth checking that it is not enabled in your kernel configuration to
+prevent any conflicts.
+
+Build the Kata Containers kernel with the previous config options, using the
+instructions described in [Building Kata Containers
+kernel](../../tools/packaging/kernel). For further details on building and
+installing guest kernels, see [the developer
+guide](../Developer-Guide.md#install-guest-kernel-images).
+
+There is an easy way to build a guest kernel that supports NVIDIA GPU:
+
+```bash
+## Build guest kernel with ../../tools/packaging/kernel
+
+# Prepare (download guest kernel source, generate .config)
+$ ./build-kernel.sh -v 5.15.23 -g nvidia -f setup
+
+# Build guest kernel
+$ ./build-kernel.sh -v 5.15.23 -g nvidia build
+
+# Install guest kernel
+$ sudo -E ./build-kernel.sh -v 5.15.23 -g nvidia install
+/usr/share/kata-containers/vmlinux-nvidia-gpu.container -> vmlinux-5.15.23-89-nvidia-gpu
+/usr/share/kata-containers/vmlinuz-nvidia-gpu.container -> vmlinuz-5.15.23-89-nvidia-gpu
+```
+
+To build NVIDIA Driver in Kata container, `kernel-devel` is required.
+This is a way to generate rpm packages for `kernel-devel`:
+
+```bash
+$ cd kata-linux-5.15.23-89
+$ make deb-pkg
+Output DEBS:
+.../linux-image-5.15.23-nvidia-gpu_5.15.23-nvidia-gpu-2_amd64.deb
+.../linux-headers-5.15.23-nvidia-gpu_5.15.23-nvidia-gpu-2_amd64.deb
+.../linux-libc-dev_5.15.23-nvidia-gpu-2_amd64.deb
+```
+
+> **Note**:
+>
+> The packages `linux-{headers,libc-dev}`  should be installed in Kata container before run NVIDIA driver
+installer is run
+>
+> Run `make rpm-pkg` to build the rpm package.
+>
+
+Before using the new guest kernel, please update the `kernel` parameters in
+ `configuration.toml`.
+
+```bash
+kernel = "/usr/share/kata-containers/vmlinuz-nvidia-gpu.container"
+```
+
+## NVIDIA GPU pass-through mode with Kata Containers
+
+Use the following steps to pass an NVIDIA GPU device in pass-through mode with Kata:
+
+1. Find the Bus-Device-Function (BDF) for GPU device on host:
+
+   ```bash
+   $ sudo lspci -nn -D | grep -i nvidia
+   0000:1d:00.0 3D controller [0302]: NVIDIA Corporation Device [10de:20b9] (rev a1)
+   0000:d0:00.0 3D controller [0302]: NVIDIA Corporation Device [10de:20b9] (rev a1)
+   ```
+
+   > PCI address `0000:d0:00.0` is assigned to the hardware GPU device.
+   > `10de:20b9` is the device ID of the hardware GPU device.
+
+2. Find the IOMMU group for the GPU device:
+
+   ```bash
+   $ BDF="0000:d0:00.0"
+   $ readlink -e /sys/bus/pci/devices/$BDF/iommu_group
+   /sys/kernel/iommu_groups/192
+   ```
+
+   The previous output shows that the GPU belongs to IOMMU group 192. The next
+   step is to bind the GPU to the VFIO-PCI driver. For easy of use we're going
+   to use the following script [vfio-pci-bind.sh](https://raw.githubusercontent.com/andre-richter/vfio-pci-bind/master/vfio-pci-bind.sh)
```

Comment:
Removed the script and added manual steps that should suffice, since we're doing some prep steps before we actually do not need the extend script features.